src/com/facebook/buck/jvm/kotlin/KotlinLibraryBuilder.java

```diff
@@ -62,7 +62,10 @@ protected CompileToJarStepFactory buildCompileStepFactory() {
       return new KotlincToJarStepFactory(
           Preconditions.checkNotNull(kotlinBuckConfig).getKotlinc(),
           extraKotlincArguments,
-          EMPTY_EXTRA_CLASSPATH);
+          EMPTY_EXTRA_CLASSPATH,
+          getJavac(),
```

Comment:
Well, that's embarrassing. We were only using mixed sources in KotlinAndroidLibrary rules so this wasn't working at all. :P
It's fixed.